authz/rbac_translator.go

```diff
@@ -283,22 +340,28 @@ func translatePolicy(policyStr string) ([]*v3rbacpb.RBAC, error) {
 	if len(policy.AllowRules) == 0 {
 		return nil, fmt.Errorf(`"allow_rules" is not present`)
 	}
+	auditLoggers, err := parseAuditLoggingOptions(policy.AuditLoggingOptions)
```

Comment:
There is a mapping from this audit condition to the conditions of deny and allow RBACs. Because we only want the same RPC to be logged at most once.

Reply:
I don't quite get what you wrote here and how it applies to this line, can you go into more detail?

Reply:
Basically, you cannot use the audit condition from the policy in both RBACs, because we need to make sure one RPC can at most be logged one time.

There is a table in the gRFC. As an example, if we want to audit on allow, then the deny RBAC should have no audit condition (I just realized right now that in fact, it doesn't even need to hold any loggers in this case), and the allow RBAC will have `ON_ALLOW`. The evaluation order is always deny -> allow, and short-circuited if applicable. 

Certainly you don't have to change this line to fix it. But this was where I realized the problem.

Reply:
So I just need some additional logic here that places the audit config in one of the two RBACs depending on the audit condition?

Reply:
That's correct.